guides/covariates_tfms.md

```diff
@@ -0,0 +1,289 @@
+We have one year (365 days) past data of our end-of-day ice-cream sales and of the average measured daily ambient temperature.
+We also noticed that our ice-cream sales depend on the day of the week so we want to include this in our model.
+
+- past target: actual past ice-cream sales `ice_cream_sales`
+- future target: predict the ice-cream sales for the next day
+- past covariates: measured average daily temperatures in the past `temperature`
+- future covariates: day of the week for past and future `weekday`
+
+Checking Table 1, a model that would accomodate this kind of covariates would be a
+`SplitCovariatesModel` (if we don't use historic values of future covariates), or
+`MixedCovariatesModel` (if we do). We choose a `MixedCovariatesModel` - the `TFTModel`.
+
+Imagine that we saw a pattern in our past ice cream sales that repeated week after week. 
+So we set `input_chunk_length = 7` days to let the model look back an entire week into the past. 
+The `output_chunk_length` can be set to `1` day to predict the next day.
+
+Now we can create a model and train it! Figure 2 shows you how `TFTModel` will use our data.
+
+```
+from darts.models import TFTModel
+
+model = TFTModel(input_chunk_length=7, output_chunk_length=1)
+model.fit(series=ice_cream_sales,
+          past_covariates=temperature, 
+          future_covariates=weekday)
+```
+
+![figure4](./images/covariates/seq_covs_single.png)
+
+**Figure 2: Overview of a single sequence from our ice-cream sales example**; Mon1 - Sun1 stand for the first 7 days from our training dataset (week 1 of the year). Mon2 is the Monday of week 2.
+
+When calling `fit()`, the models will build an appropriate `darts.utils.data.TrainingDataset`, which specifies how to slice the data to obtain training samples. If you want to control this slicing yourself, you can instantiate your own `TrainingDataset` and call `model.fit_from_dataset()` instead of `fit()`. By default, most models (though not all) will build *sequential* datasets, which basically means that all sub-slices of length `input_chunk_length + output_chunk_length` in the provided series will be used for training. 
+
+So during training, the torch models will go through the training data in sequences (see Figure 3). Using information from the **input chunk** and **output chunk**, the model predicts the future target on the output chunk. The training loss is evaluated between the predicted future target and the actual target value on the output chunk. The model trains itself by minimizing the loss over all sequences.
+
+![figure5](./images/covariates/seq_covs_1.png)
+
+**Figure 3: Prediction and loss evaluation in a single sequence**
+
+After having completed computations on the first sequence, the model moves to the next sequence, which, in the
+case of sequential models, starts one time step later than the previous sequence. This process is repeated until all 365 days were covered (see Figure 4).
```

Comment:
```suggestion
case of sequential datasets, starts one time step later than the previous sequence. This process is repeated until all 365 days were covered (see Figure 4).
```

Reply:
Actually this is not quite accurate, because loading is randomized, so the model will (in general) not see the sub-slices sequentially, but in a random order.

Reply:
True